packages/local_auth/local_auth/windows/local_auth_plugin.cpp

```diff
@@ -0,0 +1,187 @@
+    if (result_it != arguments->end()) {
+      result = std::get<T>(result_it->second);
+    }
+  }
+  return result;
+}
+
+HWND GetRootWindow(flutter::FlutterView* view) {
+  return ::GetAncestor(view->GetNativeWindow(), GA_ROOT);
+}
+
+std::wstring s2ws(const std::string& s) {
+  int len;
+  int slength = (int)s.length() + 1;
+  len = MultiByteToWideChar(CP_ACP, 0, s.c_str(), slength, 0, 0);
+  std::wstring r(len, L'\0');
+  MultiByteToWideChar(CP_ACP, 0, s.c_str(), slength, &r[0], len);
+  return r;
+}
+
+class LocalAuthPlugin : public Plugin {
+ public:
+  static void RegisterWithRegistrar(PluginRegistrarWindows* registrar);
+
+  LocalAuthPlugin(flutter::PluginRegistrarWindows* registrar);
+
+  virtual ~LocalAuthPlugin();
+
+ private:
+  flutter::PluginRegistrarWindows* registrar_;
+
+  // Called when a method is called on this plugin's channel from Dart.
+  void HandleMethodCall(const MethodCall<EncodableValue>& method_call,
+                        std::unique_ptr<MethodResult<EncodableValue>> result);
+
+  winrt::fire_and_forget Authenticate(
+      const MethodCall<EncodableValue>& method_call,
+      std::unique_ptr<MethodResult<EncodableValue>> result);
+  winrt::fire_and_forget GetAvailableBiometrics(
+      std::unique_ptr<MethodResult<EncodableValue>> result);
+  winrt::fire_and_forget IsDeviceSupported(
+      std::unique_ptr<MethodResult<EncodableValue>> result);
+};
+
+// static
+void LocalAuthPlugin::RegisterWithRegistrar(PluginRegistrarWindows* registrar) {
+  auto channel = std::make_unique<MethodChannel<EncodableValue>>(
+      registrar->messenger(), "plugins.flutter.io/local_auth",
+      &StandardMethodCodec::GetInstance());
+
+  auto plugin = std::make_unique<LocalAuthPlugin>(registrar);
+
+  channel->SetMethodCallHandler(
+      [plugin_pointer = plugin.get()](const auto& call, auto result) {
+        plugin_pointer->HandleMethodCall(call, std::move(result));
+      });
+
+  registrar->AddPlugin(std::move(plugin));
+}
+
+LocalAuthPlugin::LocalAuthPlugin(flutter::PluginRegistrarWindows* registrar)
+    : registrar_(registrar) {}
+
+LocalAuthPlugin::~LocalAuthPlugin() {}
+
+void LocalAuthPlugin::HandleMethodCall(
+    const MethodCall<EncodableValue>& method_call,
+    std::unique_ptr<MethodResult<EncodableValue>> result) {
+  if (method_call.method_name().compare("authenticate") == 0) {
+    Authenticate(method_call, std::move(result));
+  } else if (method_call.method_name().compare("getAvailableBiometrics") == 0) {
+    GetAvailableBiometrics(std::move(result));
+  } else if (method_call.method_name().compare("isDeviceSupported") == 0) {
+    IsDeviceSupported(std::move(result));
+  } else {
+    result->NotImplemented();
+  }
+}
+
+winrt::fire_and_forget LocalAuthPlugin::Authenticate(
+    const MethodCall<EncodableValue>& method_call,
+    std::unique_ptr<MethodResult<EncodableValue>> result) {
+  auto reasonW = s2ws(GetArgument<std::string>(
+      "localizedReason", method_call.arguments(), std::string()));
+
```

Comment:
You are ignoring `biometricOnly`, which would be a serious security bug for some clients. We either need to support it, or if it's not possible with the available Windows APIs, error out when it's set to true.

Reply:
I'm failing when this is true on Windows now.